### PR TITLE
BIKE Round-4 update

### PR DESCRIPTION
**Pull request by dkostic**
Update the BIKE implementation to the Round-4 spec.
The code imported from https://github.com/awslabs/bike-kem.

#1318 

* [x] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [x] Does this PR change the the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in [oqs-provider](https://github.com/open-quantum-safe/oqs-provider), [OQS-OpenSSL](https://github.com/open-quantum-safe/openssl), [OQS-BoringSSL](https://github.com/open-quantum-safe/boringssl), and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,7 +32,7 @@ set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_C_VISIBILITY_PRESET hidden)
-set(OQS_VERSION_TEXT "0.7.3-dev")
+set(OQS_VERSION_TEXT "0.8.3-dev")
```
Comment (baentsch) [inline CMakeLists.txt:35]: Please change to "0.8.0-dev"

**Review by baentsch — CHANGES_REQUESTED** [review]
Thanks very much for this contribution! Beyond the nit regarding version numbers, please also add "KEM_bike_l5" to [this list](https://github.com/dkostic/liboqs/blob/cc157293f88c44077d03c491b1ce7987a96c73b0/.CMake/alg_support.cmake#L545).

File: CMakeLists.txt (hunk 1 above)
Comment (dstebila) [inline CMakeLists.txt:35]:
```suggestion
set(OQS_VERSION_TEXT "0.8.0-dev")
```

**Comment by dstebila** [comment]
> Thanks very much for this contribution! Beyond the nit regarding version numbers, please also add "KEM_bike_l5" to [this list](https://github.com/dkostic/liboqs/blob/cc157293f88c44077d03c491b1ce7987a96c73b0/.CMake/alg_support.cmake#L545).

Done in https://github.com/open-quantum-safe/liboqs/commit/1eec2e4d5ee07aaf55083722a5839a38bc8c84fe

**Review by dstebila — APPROVED** [review]
Ready to merge once builds pass

**Comment by dstebila** [comment]
@dkostic looks like there are a few nits picked up by scan-build.  Are you able to see them in the CI output at https://app.circleci.com/pipelines/github/open-quantum-safe/liboqs/2404/workflows/097d86fc-1e44-4360-95c5-a94441e9835d/jobs/19355 ?

**Comment by dkostic** [comment]
Hi, sorry for the delay, I was out last week. I tried to fix the CI unused variable warning, let's see if it works.

**Comment by dkostic** [comment]
Is there a way for me to see the CI failures? The link Douglas sent works for me, but when I try to access the logs from github it tells me I'm unauthorized.

**Comment by dstebila** [comment]
> Is there a way for me to see the CI failures? The link Douglas sent works for me, but when I try to access the logs from github it tells me I'm unauthorized.

I've added you as a contributor so you should be able to access it. The build failure you just had was that some CI jobs only run if they come from team members, but I've triggered a re-run.

**Comment by dkostic** [comment]
Thanks Douglas!

**Review by crockeea — APPROVED** [review]

**Review by baentsch — APPROVED** [review]

**Review by dstebila — APPROVED** [review]

**Comment by baentsch** [comment]
@dkostic @dstebila Would you consider the code in this PR to be correctly tagged as "NIST round 3" and "v4.1" (see https://github.com/open-quantum-safe/liboqs/blob/40b01fdbb270f8614fde30e65d30e9da18c02393/docs/algorithms/kem/bike.yml#L23-L24) or should these tags be upgraded to read "4" and "5.1" respectively? Please check the resultant documentation for correctness, too: https://github.com/open-quantum-safe/liboqs/blob/main/docs/algorithms/kem/bike.md.
FYI, this information is used in all downstream projects to correctly tag the constituent algorithm versions (incl. hybrids, etc.). Please check whether anything else is not quite right (ancestor? what else?).

If an update is required, please either do it yourself or let me know whether I can/shall proceed as per the simple proposal above.

**Pull request merged**